custom-mao.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+# This script disables the Machine API Operator deployed by the Cluster Version
+# Operator and subsittutes a locally-built one. This is also a prerequisite to
+# deploying a custom actuator (e.g. a custom cluster-api-provider-baremetal),
+# by linking to a different quay.io image in the machine-api-operator's
+# images.json file.
```

Comment:
I think it's actually possible to deploy a custom CAPBM without the custom MAO.

You still have to tell the CVO to stop managing the MAO.  After that though, you should be able to edit the images ConfigMap and restart the MAO.  That should be a simpler process if the use case is only a custom CAPBM.

Reply:
I'm testing this out (stop managing the machine-api-operator deployment from CVO, update the machine-api-operator-images configmap, delete the machine-api-operator and machine-api-controllers pods), but something keeps reverting the configmap. @russellb do you happen to know what that is and how to make it stop?

Reply:
My guess is that it's the CVO doing it.  CVO owns the ConfigMap.  The easy way to avoid this is to stop the CVO completely.  I'm guessing you could also configure it to stop managing the ConfigMap, similar to the patch that tells it to stop managing the Deployment of the machine-api-operator.

Reply:
I found this documentation helpful to set the CVO to stop managing a configmap or to disable it altogether. https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/clusterversion.md#setting-objects-unmanaged

Reply:
Thanks, that worked but it turns out that after changing the contents of the configmap there's no way to prod the deployment to update (at least until k8s 1.15). So I ended up just creating new configmap and pointing the deployment at that instead, which works nicely (although it always takes exactly 119s to kick in for some reason).